client/components/app.jsx

```diff
@@ -124,7 +123,7 @@ export default class App extends React.Component {
         <>
           <Header cartItemCount={this.state.cart.length} setView={this.setView} />
           <Transition key={this.state.view.name}>
-            <CartSummary cart={this.state.cart} setView={this.setView} total={this.total} deleteFromCart={this.deleteFromCart} />
+            <CartSummary cart={this.state.cart} setView={this.setView} total={this.calculateTotal} deleteFromCart={this.deleteFromCart} />
```

Comment:
Better to pass the component what it really wants (the total). so you can call this function here.